src/control/server/harness.go

```diff
@@ -255,24 +256,26 @@ func (h *IOServerHarness) waitInstancesReady(ctx context.Context) error {
 	return nil
 }
 
-// monitorInstances listens for exit results from instances or harness and will
-// return only when all harness instances are stopped and restartInstances
+// monitor listens for exit results from instances or harness and will
+// return only when all harness instances are stopped and restart
 // signal is received.
-func (h *IOServerHarness) monitorInstances(ctx context.Context) error {
+func (h *IOServerHarness) monitor(ctx context.Context) error {
 	h.log.Debug("monitoring instances")
 	for {
 		select {
 		case <-ctx.Done(): // received when harness is exiting
-			h.log.Debug("harness exiting")
 			return ctx.Err()
 		case err := <-h.errChan: // received when instance exits
 			// TODO: Restart failed instances on unexpected exit.
 			allInstancesStopped := !h.HasStartedInstances()
 			msg := fmt.Sprintf("instance exited: %v", err)
 			if allInstancesStopped {
 				msg += ", all instances stopped!"
+				h.setRestartable()
```

Comment:
Kind of curious why restartable is its own member - looks like it is equivalent to !h.HasStartedInstances(). Is there a race condition or some other complication in having h.IsRestartable() just call h.HasStartedInstances()?

Reply:
This so tried to describe in the commit message:
"restartable atomic prevents restart between harness start and
instance start "
basically there is a window between when the harness has been started and when the instances get started which would provide the situation for a "HarnessRestart" gRPC said to be accepted out of turn.

Reply:
Thanks for the explanation.